src/router/utils/HelperFunctions.jsx

```diff
@@ -98,3 +116,83 @@ export const removeFromLikedList = async (video, videoDispatch) => {
     console.log(error);
   }
 };
+
+export const addToHistory = async (video, videoDispatch) => {
+  try {
+    const response = await axios.post(
+      "/api/user/history",
+      { video },
+      {
+        headers: {
+          authorization: userToken,
+        },
+      }
+    );
+    if (response.status === 201) {
+      videoDispatch({ type: "ADD_TO_HISTORY", payload: response.data.history });
+    }
+  } catch (error) {
+    Toast({
+      type: "error",
+      message: "Oops! An error occurred.Try again later.",
+    });
+    console.log(error);
+  }
+};
+
+export const deleteFromHistory = async (video, videoDispatch) => {
+  try {
+    const response = await axios.delete(`/api/user/history/${video._id}`, {
+      headers: {
+        authorization: userToken,
```

Comment:
Suggestion: instead of passing headers explicitly in each request, you can setup it in axios inteceptors. That way, you won't need to passed headers in each request

Reply:
I will read about this and implement it in the end. Thank you!